### PR TITLE
Fix error being overridden incorrectly on FS.Rename

### PR DESCRIPTION
**Pull request by kjpou1**
While renaming a file and if one of the parent paths do not exist the error code is being overwritten to EBUSY instead of reporting the actual error:

```
Error no 44 Error: ENOENT Message: No such file or directory 

```

Closes https://github.com/emscripten-core/emscripten/issues/11804

### CODE REVIEW

File: src/library_fs.js [hunk 1]
```diff
@@ -726,6 +726,9 @@ FS.staticInit();` +
         lookup = FS.lookupPath(new_path, { parent: true });
         new_dir = lookup.node;
       } catch (e) {
+        // do not override failing error
```
Comment (sbc100) [inline src/library_fs.js:729]: What if we just remove this try/catch completely?

I guess we should also a test (or modify an existing test)?

Comment (kjpou1) [reply src/library_fs.js:729]: Do not know what that will do if we remove it for other error messages.  Just let me know if you would prefer to remove the try catch.

What tests are there for that and where so I can change it.

Comment (kjpou1) [reply src/library_fs.js:729]: There are tests as part of the file system tests that this fixes.

Comment (kripken) [reply src/library_fs.js:729]: It doesn't look like we have a test for returning EBUSY here - nothing comes up in a search, and removing the try-catch as @sbc100 suggests passes the test suite.

Reading the docs on `rename()`, I don't think we should be returning EBUSY. So I think that's the proper fix, unless we have other information?

Comment (sbc100) [reply src/library_fs.js:729]: Looks like this failure case should probably be added to `tests/stdio/test_rename.c`

Comment (kjpou1) [reply src/library_fs.js:729]: So remove the try catch and just let it fail and pass the error percolate up is the way to fix this right.

Comment (sbc100) [reply src/library_fs.js:729]: To me that look like a better solution yes... I can't see why use the EBUSY thing is below...  but I don't know this code.

Best thing is to upload that patch and see if any tests fail.

Comment (kjpou1) [reply src/library_fs.js:729]: Will do.  Thanks

Comment (kjpou1) [reply src/library_fs.js]: Addressed in recent commit

**Comment by kjpou1** [comment]
Test added

**Comment by sbc100** [comment]
Looks like you have a conflict. 

While you are there can you add a comment to the test code saying why that test case is needed.    e.g. "test that non-existant parent during rename generates the correct error code"?

**Pull request merged**